pkg/controlplane/apiserver/config.go

```diff
@@ -159,13 +166,38 @@ func BuildGenericConfig(
 		genericConfig.FlowControl, lastErr = BuildPriorityAndFairness(s, clientgoExternalClient, versionedInformers)
 	}
 
+	if reconcilers.PeerProxyFeatureEnabled() {
+		genericConfig.PeerEndpointReconciler = createPeerEndpointLeaseReconciler(*genericConfig, storageFactory)
+		genericConfig.PeerProxy, lastErr = BuildPeerProxy(versionedInformers, genericConfig.StorageVersionManager, s.ProxyClientCertFile,
+			s.ProxyClientKeyFile, s.GenericServerRunOptions.PeerCAFile, s.GenericServerRunOptions.PeerBindAddress)
+	}
+
 	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.AggregatedDiscoveryEndpoint) {
 		genericConfig.AggregatedDiscoveryGroupManager = aggregated.NewResourceManager("apis")
 	}
 
 	return
 }
 
+// createPeerEndpointLeaseReconciler creates a clone of EndpointReconciler to set up a separate apiserver endpoint
+// reconciliation loop that is used to find network locations of peer apiservers for peer proxy
+func createPeerEndpointLeaseReconciler(c genericapiserver.Config, storageFactory serverstorage.StorageFactory) reconcilers.PeerEndpointReconciler {
+	endpointClient := corev1client.NewForConfigOrDie(c.LoopbackClientConfig)
+	endpointSliceClient := discoveryclient.NewForConfigOrDie(c.LoopbackClientConfig)
+	endpointsAdapter := reconcilers.NewEndpointsAdapter(endpointClient, endpointSliceClient)
+
+	ttl := controlplane.DefaultEndpointReconcilerTTL
+	config, err := storageFactory.NewConfig(api.Resource("apiServerIPInfo"))
+	if err != nil {
+		klog.Fatalf("Error creating storage factory config: %v", err)
```

Comment:
I think is better to return an error and handle it in the caller instead of panicing here

Reply:
returning error now.